examples/swift/hello_world/ViewController.swift

```diff
@@ -34,47 +40,38 @@ final class ViewController: UITableViewController {
   }
 
   private func performRequest() {
-    self.requestCount += 1
-    let requestID = self.requestCount
-    var request = URLRequest(url: kURL)
-    request.addValue("s3.amazonaws.com", forHTTPHeaderField: "host")
-
-    // Note that the request is sent to the envoy thread listening locally on port 9001.
-    NSLog("Starting request to '\(kURL.path)'")
-    self.session.dataTask(with: request) { [weak self] data, response, error in
-      DispatchQueue.main.async {
-        self?.handle(response: response, with: data, error: error, id: requestID)
-      }
-    }.resume()
-  }
-
-  private func handle(response: URLResponse?, with data: Data?, error: Error?, id: Int) {
-    if let error = error {
-      return self.add(result: .failure(RequestError(id: id, message: "\(error)")))
-    }
-
-    guard let response = response as? HTTPURLResponse, let data = data else {
-      return self.add(result: .failure(RequestError(id: id, message: "missing response data")))
+    guard let envoy = self.envoy else {
+      NSLog("Failed to start request - Envoy is not running")
+      return
     }
 
-    guard response.statusCode == 200 else {
-      let error = RequestError(id: id, message: "failed with status \(response.statusCode)")
-      return self.add(result: .failure(error))
-    }
-
-    guard let body = String(data: data, encoding: .utf8) else {
-      return self.add(result: .failure(RequestError(id: id, message: "failed to deserialize body")))
-    }
+    self.requestCount += 1
+    NSLog("Starting request to '\(kRequestPath)'")
 
-    let untypedHeaders = response.allHeaderFields
-    let headers = Dictionary(uniqueKeysWithValues: untypedHeaders.map
-    { header -> (String, String) in
-      return (header.key as? String ?? String(describing: header.key), "\(header.value)")
-    })
+    let requestID = self.requestCount
+    let request = RequestBuilder(method: .get, scheme: kRequestScheme,
+                                 authority: kRequestAuthority,
+                                 path: kRequestPath).build()
+    var status = -1
```

Comment:
This is pretty lame to do in order to be able to log/read the status code of a given error and success -- especially for the unary case.

Reply:
Yea agreed - I think we should discuss providing a way for `onError` to pipe through the status code as well (or maybe introduce a new stream handler)

Reply:
Actually, thinking about this more - `onError` here will be called if an error occurs within Envoy (not if a 4xx/5xx is returned). So in that case, seems like we should return an `EnvoyError` in that lambda and not worry about the HTTP status code being piped through there. I can update the logs in this example here too. WDYT?